semgrep-core/parsing/Parse_csharp_tree_sitter.ml

```diff
@@ -70,6 +70,87 @@ let var_def_stmt (decls : (entity * variable_definition) list) (attrs : attribut
   ) decls in
   stmt1 stmts
 
+type direction =
+  | Ascending
+  | Descending
+
+and linq_query_part = 
```

Comment:
Nice. We might put that at some point in AST_generic.ml instead, especially if another programming
language has a similar feature (e.g., F#, but Python has also some List comprehensions which are a bit
similar, even though far more limited).